File: payment_gateway/src/main/java/com/ftn/uns/payment_gateway/model/Magazine.java
package com.ftn.uns.payment_gateway.model;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
public class Magazine {

    @Id
    @Column(length = 8, nullable = false, unique = true)
    private String issn;

    private String title;

    @OneToMany
    @JoinColumn(name = "MAGAZINE")
    private Set<PaymentServiceDetails> details = new HashSet<>();

    public Magazine() {
    }

    public Magazine(String issn, String title, Set<PaymentServiceDetails> details) {
        this.issn = issn;
        this.title = title;
        this.details = details;
    }

    public String getIssn() {
        return issn;
    }

    public void setIssn(String issn) {
        this.issn = issn;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Set<PaymentServiceDetails> getDetails() {
        return details;
    }

    public void setDetails(Set<PaymentServiceDetails> details) {
        this.details = details;
    }
}

File: payment-gateway-front/src/app/model/magazine.ts
import { PaymentDetails } from "./payment-details";

export class Magazine {
    constructor(
        public issn: string,
        public title: string,
        public details: PaymentDetails[]
    ) {}

    public print(): String{
        let str = "ISSN: ISSN "+this.issn[0]+this.issn[1]+this.issn[2]+this.issn[3]+'-'+this.issn[4]+this.issn[5]+this.issn[6]+this.issn[7] 
            + "\nTitle: "+this.title
            +"\nTypes: ";
        for(let detail of this.details){
            str += detail.type + ",";
        }

        return str;
    }
}
